src/elements/content-uploader/UploadStateV2.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @flow
+ * @file Upload state component
+ */
+
+import React from 'react';
+import classNames from 'classnames';
+import { FormattedMessage } from 'react-intl';
+import ErrorEmptyState from '../../icons/states/ErrorEmptyState';
+import UploadIllustration from '../../icons/illustrations/UploadIllustration';
+import UploadSuccessState from '../../icons/states/UploadSuccessState';
+import messages from '../common/messages';
+import Browser from '../../utils/Browser';
+import UploadStateContent from './UploadStateContent';
+import { VIEW_ERROR, VIEW_UPLOAD_EMPTY, VIEW_UPLOAD_IN_PROGRESS, VIEW_UPLOAD_SUCCESS } from '../../constants';
+
+import './UploadState.scss';
+import './UploadStateV2.scss';
+
+type Props = {
+    canDrop: boolean,
+    hasItems: boolean,
+    isFolderUploadEnabled: boolean,
+    isOver: boolean,
+    isTouch: boolean,
+    onSelect: Function,
+    view: View,
+};
+
+const UploadStateV2 = ({ canDrop, hasItems, isOver, isTouch, view, onSelect, isFolderUploadEnabled }: Props) => {
+    let icon;
+    let content;
+    switch (view) {
+        case VIEW_ERROR:
+            icon = <ErrorEmptyState />;
+            content = <UploadStateContent message={<FormattedMessage {...messages.uploadError} />} />;
+            break;
+        case VIEW_UPLOAD_EMPTY:
+            if (!hasItems) {
+                icon = isOver ? <UploadIllustration width={200} height={200} /> : <UploadIllustration />;
+            }
+            /* eslint-disable no-nested-ternary */
+            content =
+                canDrop && hasItems ? (
+                    <UploadStateContent message={<FormattedMessage {...messages.uploadInProgress} />} />
+                ) : isTouch ? (
```

Comment:
way too much nesting going on, maybe break it into functions to reduce arrowness ====>